### ISSUE TITLE
Arbitrary instance for Image is problematic

### ISSUE THREAD

**Issue by ayberkt** [CIFASIS/QuickFuzz#12]
The `Arbitrary` instance declaration for `Image PixelRGBA8` is as follows:

``` haskell
instance Arbitrary (Image PixelRGBA8) where
   arbitrary = do
       l <- listOf (arbitrary :: Gen Word8)
       w <- (arbitrary :: Gen Int)
       h <- (arbitrary :: Gen Int)
       return $ Image { imageWidth = w, imageHeight = h, imageData = VS.fromList l }
```

The width, height, and the length of the data vector for the image have to be consistent with each other. This declaration results in QuickCheck causing an `OutOfBounds` exception. Moreover width and height have to be constrained to have positive values, which is trivial to fix. But I am not sure how to keep image parameters consistent with each other. One quick fix I implemented was to arbitarily choose `l` and `w` and then set `h` based on those two. But then the image array has to be constrained to have a length divisible by 4 (in the case of PixelRGBA8). I am not really experienced with QuickCheck so there probably is a better way of doing this.

**Comment by gaa-cifasis** [comment]
Hi,

Sorry for the delay. Indeed this quick and dirty arbitrary instance can be improve and generalized for different types of pixels. You also mentioned that the imageData should be restricted to have certain size. This can be used if you want to test code that already enforces such condtion, but it is also interesting to know if you have more or less data that you specified in the headers. Anyway, we could generate only this kind of "valid" images. Let me do some test and i will report back if this was useful or not in the bug hunting.

In the meantime, feel free to make your own research and submit a example code or PR. Thanks a lot for your feedback!

**Comment by gaa-cifasis** [comment]
Fixed c3aeef4a5f18b6473260abad9eb25cda9f3833dd. Thanks again.
